clang/test/SemaSYCL/check-notdirect-attribute-propagation.cpp

```diff
@@ -10,15 +10,15 @@
 }
 
 #else
-[[cl::reqd_work_group_size(2, 2, 2)]] void not_direct_two() {} // expected-note {{conflicting attribute is here}}
```

Comment:
Note that the changes here are because we are doing a DFS instead of a BFS.  This obviously does not change the validity of any program.

IMO, the diagnostics are better now, since you are more likely to have the first diagnostic be 'child of the thing that added it' rather than a peer, which seems easier to debug as an end user to me.  AND as added benefit, we only end up walking the call graph 1x instead of 2x.